deployment/clouddeploy/deployer.go

```diff
@@ -1462,6 +1470,101 @@ func (p *Deployer) GetCertificate(ctx context.Context, clusterID string) (string
 	return strings.TrimSpace(lastCert.Pem), nil
 }
 
+func (d *Deployer) startLogCollection(ctx context.Context, cluster *clusterInfo) error {
+	var startCollectingServerLogsRequest = &capellacontrol.StartCollectingServerLogsRequest{
+		HostName: cbdcconfig.DEFAULT_HOST_NAME,
+	}
+
+	var err = d.mgr.Client.StartCollectingServerLogs(ctx, cluster.Cluster.Id, d.internalSupportToken,
+		startCollectingServerLogsRequest)
+
+	if err != nil {
+		errors.Wrap(err,
+			fmt.Sprintf("failed to start server log collection: %s", err))
+	} else {
+		d.logger.Info(fmt.Sprintf("Log collection have started for cluster: %s", cluster.Cluster.Id))
+	}
+
+	return err
+}
+
+func (d *Deployer) downloadFromURL(url, destPath string) error {
+	// Ensure the directory exists
+	if err := os.MkdirAll(destPath, 0755); err != nil {
+		return err
+	}
+
+	filePath := filepath.Join(destPath, filepath.Base(url))
+	logFile, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	defer logFile.Close()
+	// Perform the HTTP GET request to download the file
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Check if the response status code indicates success
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("server returned non-200 status code: %d", resp.StatusCode)
+	}
+
+	//// Copy the response body to the file
+	_, err = io.Copy(logFile, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *Deployer) CollectLogs(ctx context.Context, clusterID string, destPath string) ([]string, error) {
+	cluster, err := d.getCluster(ctx, clusterID)
+	if err != nil {
+		return []string{}, err
+	}
+
+	err = d.startLogCollection(ctx, cluster)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var downloadServerLogsRequest = &capellacontrol.DownloadServerLogsRequest{
+		HostName: cbdcconfig.DEFAULT_HOST_NAME,
```

Comment:
Related to my comment above about configurability, this should probably be configurable.

Reply:
update code to make it configurable